scripts/format_app.py

```diff
@@ -0,0 +1,70 @@
+# format_app.py
+
+import argparse
+import os
+import subprocess
+import sys
+
+
+def handle_cli_args():
+  parser = argparse.ArgumentParser(description='A script for formatting a given app within emqx. Example Usage: /bin/python3.9 scripts/format_app.py -a emqx_resource -b format_resource_app')
+  parser.add_argument("-a", "--application", help="The application which is to be formatted.")
+  parser.add_argument('-b', "--branch", help="The git branch to be switched to before formatting the code. Required unless the -f option is passed in which case the value will be ignored even if provided.")
+  parser.add_argument('-f', "--format_in_place", default=False, action="store_true", help="Pass the -f option to format on the current git branch, otherwise a branch name must be provided by passing the -b option.")
+  args = parser.parse_args()
+
+  if args.format_in_place == False and args.branch == None:
+    sys.exit("A new git branch name must be provided with the -b option unless the -f option is given to format the code in place.")
+
+  return args
+
+
+def get_app_path(application):
+  root_path = os.path.dirname(os.path.realpath(__file__))
+  full_path = f"{root_path}/../apps/{application}"
+  isdir = os.path.isdir(full_path) 
+
+  if isdir:
+    return full_path
+  
+  sys.exit(f"The application provided ({application}) does not appear at the expected location: {full_path}")
+
+
+def maybe_switch_git_branch(format_in_place, branch):
+  if not format_in_place:
+    PIPE = subprocess.PIPE
+    process = subprocess.Popen(["git", "status"], stdout=PIPE, text=True)
+    stdoutput, stderroutput = process.communicate()
+
+    if f"On branch {branch}" in stdoutput:
+      return
+    elif "working tree clean" in stdoutput:
+      subprocess.call(["git", "checkout", "-b", branch])
+    else:
+      sys.exit("Cannot switch git branches while there are changes waiting to be committed.")
+
+
+def maybe_add_rebar_plugin():
+  with open("rebar.config", "r+") as f:
+    text = f.read()
+    
+    if not "{project_plugins, [erlfmt]}." in text:
+      f.write("\n{project_plugins, [erlfmt]}.\n")
```

Comment:
Maybe it would be safer to exit with an error and tell the user to add it manually, since there might be other plugins, or the formatting of the line might be slightly off the expected.